test-framework/common/src/main/java/io/quarkus/test/common/JarLauncher.java

```diff
@@ -39,6 +40,7 @@ private JarLauncher(Path jarPath, Config config) {
                 config.getValue("quarkus.http.test-port", OptionalInt.class).orElse(DEFAULT_PORT),
                 config.getValue("quarkus.http.test-ssl-port", OptionalInt.class).orElse(DEFAULT_HTTPS_PORT),
                 config.getValue("quarkus.test.jar-wait-time", OptionalLong.class).orElse(DEFAULT_JAR_WAIT_TIME),
+                config.getOptionalValue("quarkus.test.argLine", String.class).orElse(null),
```

Comment:
This needs to be added to `io.quarkus.deployment.dev.testing.TestConfig` so it ends up in the 'all properties' documentation. 

It would also be good to include this in the testing documentation, as we don't have any info at the moment on doing code coverage with @QuarkusIntegrationTest.

Reply:
Alright. So wrt the `io.quarkus.deployment.dev.testing.TestConfig` I'm a bit confused. 
I see in that class 
```
    /**
     * The profile to use when testing the native image
     */
    @ConfigItem(defaultValue = "prod")
    String nativeImageProfile;
```
Which seems a bit misleading, since it's not `prod` explicitly as a default -- but rather `null`.

So, while I patterned the implementation of this OptionalValue off that, I feel like I probably should _not_ define a `defaultValue` for the `@ConfigItem`, does that sound right?

Reply:
wrt the testing documentation -- where would I find that, and I'm not convinced I'm your guy for documenting that.

I've found the testing documentation to be a bit ... lacking... especially in the realm of Integration testing, and what-not.

Reply:
Yea, the default value should be null. Also the property name should be something line `quarkus.test.integration-jvm-arg-line` to fit in with our naming convention, and also to make it clear where this applies so you have something like:

```
    /**
     * JVM parameters that are used to launch jar based integration tests.
     */
    @ConfigItem
    String integrationJvmArgLine;

```

Reply:
The documentation is at: https://github.com/quarkusio/quarkus/blob/main/docs/src/main/asciidoc/getting-started-testing.adoc#16-using-quarkusintegrationtest

Reply:
Documentation and TestConfig updated. Let me know what else I can do, or any other feedback you have.

Reply:
@stuartwdouglas I'm not sure why this is failing, it's also failing locally.
It looks like I needed to add this ConfigItem to other classes as well, but I can't quite seem to put my finger on it.

Reply:
Oh wow. How'd I miss that it needs to be `Optional<String>`? :facepalm: